### ISSUE TITLE
rewrite gidwrapper

### ISSUE THREAD

**Issue by ezrizhu** [binpash/try#143]
Current we have https://github.com/ezrizhu/gidmapper - we want to rewrite this in C for easier installation.

**Comment by mgree** [comment]
Rewriting it in C is not the important part. The main thing we have to do is understand whether or not such a gidmapper is safe and---if not---what we can do instead.

**Comment by SleepyMug** [comment]
My understanding is that `unshare` command internally forks into two processes (for pid namespace). And when I was re-implementing that logic in Rust I can set uid/gid mappings from the parent process without needing additional utility.

**Comment by ezrizhu** [comment]
> My understanding is that `unshare` command internally forks into two processes (for pid namespace). And when I was re-implementing that logic in Rust I can set uid/gid mappings from the parent process without needing additional utility.

that's correct, the main reason why gidmapper exists because we can't give a shell script the setgid cap.

**Comment by mgree** [comment]
We can just use `newuidmap` and `newgidmap` to do what we want, but #221 indicates that we might be out of luck here.
